northstar/src/runtime/process/init.rs

```diff
@@ -30,11 +30,130 @@ pub(super) struct Init {
     pub seccomp: Option<AllowList>,
 }
 
+/// This is the type used to communicate with the init process from the runtime.
+#[derive(Debug, Serialize, Deserialize)]
+pub enum InitMessage {
+    NewChild { pid: i32 },
+    DeadChild { pid: i32, exit_status: ExitStatus },
+}
+
+/// Init commands
+#[derive(Debug, Serialize, Deserialize)]
+pub enum InitApi {
+    /// Executes an command
+    Exec {
+        path: CString,
+        args: Vec<CString>,
+        env: Vec<CString>,
+    },
+}
+
+/// Init process main function
+fn init_main(
+    mut rt_tx: channel::Sender<InitMessage>,
+    mut rt_rx: channel::Receiver<InitApi>,
+    seccomp: Option<AllowList>,
+) -> ! {
+    enum Payload {
+        ChildExit { pid: i32, exit_status: ExitStatus },
+        Command(InitApi),
+    }
+    let (tx, rx) = mpsc::channel();
+
+    // This thread forwards the commands coming from the runtime
+    let command_tx = tx.clone();
+    let _command_thread = std::thread::spawn(move || loop {
+        let command = rt_rx
+            .recv()
+            .expect("Failed to receive init command from runtime");
+        command_tx.send(Payload::Command(command)).unwrap();
+    });
+
+    // This thread sends the exit status of the child processes
+    let exit_status_tx = tx;
```

Comment:
This just a rename.

Reply:
yeah, its just some verbosity to avoid wandering where the `tx` came from inside the thread. Can remove if if you want.